packages/contract/src/lib.rs

```diff
@@ -0,0 +1,371 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+pub use near_sdk::serde_json::{self, json, Value};
+use near_sdk::{
+    borsh,
+    collections::LookupMap,
+    env, ext_contract,
+    json_types::U128,
+    log, near_bindgen,
+    serde::{Deserialize, Serialize},
+    AccountId, BorshStorageKey, Gas, PanicOnDefault, Promise, PromiseResult,
+};
+
+#[global_allocator]
+static ALLOC: near_sdk::wee_alloc::WeeAlloc = near_sdk::wee_alloc::WeeAlloc::INIT;
+
+pub const FRACTIONAL_BASE: u128 = 10_000;
+pub const BASE_GAS: Gas = 5_000_000_000_000;
+
+#[ext_contract(token_contract)]
+trait FungibleToken {
+    fn ft_transfer(receiver_id: String, amount: U128, memo: String);
+}
+
+#[ext_contract(ext_self)]
+trait LogInfo {
+    fn log_transfer(
+        receiver: String,
+        amount: U128,
+        token: AccountId,
+        sender: AccountId,
+        sender_discord: String,
+        receiver_discord: String,
+        server_discord: String,
+    );
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct FtTransferCallMsg {
+    receiver: String,
+    sender_discord: String,
+    receiver_discord: String,
+    server_discord: String,
+}
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct ServerCoinUser {
+    server: String,
+    coin: String,
+    user: String,
+}
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct ServerCoin {
+    server: String,
+    coin: String,
+}
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct RankingItem {
+    user: String,
+    amount: U128,
+}
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct LastTransactionsItem {
+    sender: String,
+    receiver: String,
+    coin: String,
+    amount: U128,
+}
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+pub struct Contract {
+    pub owner_id: AccountId,
+    pub user_history: LookupMap<ServerCoinUser, u128>,
+    pub server_ranking: LookupMap<ServerCoin, Vec<RankingItem>>,

```

Comment:
Size is not fixed. It has a maximum size of 5, but starts out as an empty vector.
It would be possible to implement using an array, but I cannot see any benefits. A Vec uses an array of fixed size in the background, memory usage is very similar and provides a more readable API.